### DISCUSSION TITLE
Ubuntu revert all changes made in bash session

### DISCUSSION THREAD

**Discussion by menih** [microsoft/WSL#6497] [General]
I've upgraded to WSL 2 following the instruction provided. Since than, any changes I apply to files on subsystem are GONE!
I tried converting ubuntu 18 to v2. No errors. But does not fix the issue either.
I tried both reset and repair Ubuntu installation. Did not help.
I've installed 20.04 LTS and removed 18.04 LTS. SAME ISSUE. I ended up loosing all my data also... aghghghg

I am lost. Is this a known issue? How do I fix this...

**Comment by menih** [comment]
I should be specific about the file that is being reverted - /etc/hosts. I added an entry.

Also forgot to mention, session gets closed prematurely without any trace... 
'bash' command does not launch ubuntu.

I think my whole subsystem is in a real bad shape. But I don't know best way to reset it now...

**Comment by ghost** [comment]
Hello,

I have been interested in WSL2 recently. I think I can help you with some issues.

## File Recovery on WSL
I know a way to access files through the NTFS file system. Open Windows Explorer and enter the following expression:

```
\\wsl$
```

![01](https://user-images.githubusercontent.com/57282885/104821844-c70b4f80-584f-11eb-99c9-3ae163f8f716.png)
<div align="center">
 Picture 1. Entering \\wsl$ in the windows explorer
</div>

![02](https://user-images.githubusercontent.com/57282885/104821851-cd99c700-584f-11eb-8ffb-1c2d0d6b078e.png)
<div align="center">
Picture 2. Selecting the Linux distribution to recover the file
</div>

![03](https://user-images.githubusercontent.com/57282885/104822324-ae506900-5852-11eb-8d62-04d62f33e67e.png)
<div align="center">
Picture 3. Copying files/projects to be recovered to the NTFS file system
</div>

This way, you can access the project files you are working on. You can use [VS Code Studio](https://code.visualstudio.com/docs/remote/wsl) and [Git](https://docs.microsoft.com/en-us/windows/wsl/tutorials/wsl-git) when working with WSL to avoid data loss. 

You can check the WSL version assigned to each of the Linux distributions you have installed by opening the PowerShell command line and entering the command (only available in Windows Build 18362 or higher): (**Resource:** [Set your distribution version to WSL 1 or WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10#set-your-distribution-version-to-wsl-1-or-wsl-2))

```powershell 
wsl --list --verbose
```

![05](https://user-images.githubusercontent.com/57282885/104823547-fcb63580-585b-11eb-9124-3d79d233e695.png)
<div align="center">
Picture 4. Ubuntu is running on WSL 2.
</div>

To set a distribution to be backed by either version of WSL please run: (**Resource:** [Set your distribution version to WSL 1 or WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10#set-your-distribution-version-to-wsl-1-or-wsl-2))

```powershell 

wsl --set-version <distribution name> <versionNumber>

# Use Case: https://github.com/microsoft/WSL/discussions/5645#discussioncomment-135169

```

I think the Ubuntu 18.04 user directory will not be deleted during the system upgrade (Ubuntu 18.04 > Ubuntu 20.04 / WSL 1 > WSL 2). You can change the default WSL distribution: 

```powershell
# Convert an existing WSL1 distro to WSL2 from PowerShell
wsl --set-version Ubuntu 2
```

![06](https://user-images.githubusercontent.com/57282885/104823625-8cf47a80-585c-11eb-88b0-722e9eff850f.png)
<div align="center">
Picture 5. Ubuntu 20.04 will be run on WSL2
</div>

## Bash Session Records

You can view the **.bash_history** log in your user directory to see the last commands used on the command line. **.bash_history** is a text file that can be opened in notepad. 

![04](https://user-images.githubusercontent.com/57282885/104822901-06896a00-5857-11eb-83e9-1b4af983d04d.png)
<div align="center">
Picture 6. .bash_history command history log
</div>

## Uninstall WSL

WSL distributions installed from the store can be uninstalled like any other Windows app, by right-clicking on the app tile and clicking Uninstall, or via PowerShell using the Remove-AppxPackage cmdlet. [Resource: How do I uninstall a WSL Distribution?](https://docs.microsoft.com/en-us/windows/wsl/faq#:~:text=How%20do%20I%20uninstall%20a%20WSL%20Distribution%3F&text=WSL%20distributions%20installed%20from%20the,using%20the%20Remove%2DAppxPackage%20cmdlet.)

Open command prompt with administrator rights and run the following command: 

```bash 
lxrun /uninstall /full
```

## Reinstall WSL

You can follow the manual steps in the [document](https://docs.microsoft.com/en-us/windows/wsl/install-win10#manual-installation-steps) during the reinstallation phase. When you run PowerShell during the setup again, it should have administrator permissions.
